### PR TITLE
Add Comparing#mapper

### PR DESCRIPTION
**Pull request by luanpotter**
When you already have a Comparator, you can use `Comparing#mapper` to combine it with a new mapping function. It takes a `Comparator<T>` and a mapping of `V -> T` to create a `Comparator<V>`.

### CODE REVIEW

File: lib/comparing.dart [hunk 1]
```diff
@@ -12,6 +12,14 @@ class Comparing {
     return (a, b) => mapper(a).compareTo(mapper(b));
   }
 
+  /// Use if you already have a custom comparator you wish to use but your list needs some mapping to get to the comparator's type.
```
Comment (spydon) [inline lib/comparing.dart]: Very long line

**Review by spydon — APPROVED** [review]

File: lib/comparing.dart (hunk 1 above)
Comment (luanpotter) [reply lib/comparing.dart]: I dont think this file respects any line limit for comments :P 

what is our suggested max so I can wrap everything?

Comment (luanpotter) [reply lib/comparing.dart]: I wrapped at 120, I think that is reasonable?

Comment (spydon) [reply lib/comparing.dart]: I think comments, dartdoc and code should all go at the same line length, which we now have as the old school 80. It becomes easier to read and size containing windows that doesn't wrap the lines.

Comment (luanpotter) [reply lib/comparing.dart]: wrapped all files at 80!

**Pull request merged**